<!DOCTYPE html><html lang="en"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=5"><title>GuoJiayi</title><meta name="author" content="Guo Jiayi"><link rel="shortcut icon" href="/img/favicon.png"><link rel="stylesheet" href="/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.13.0/css/all.min.css"><link rel="stylesheet" href="/css/custom.css"><meta name="generator" content="Hexo 7.3.0"></head><body><header id="page_header"><div class="header_wrap"><div id="blog_name"><a class="blog_title" id="site-name" href="/">GuoJiayi</a></div><button class="menus_icon"><div class="navicon"></div></button><ul class="menus_items"><li class="menus_item"><a class="site-page" href="/about/"> About</a></li><li class="menus_item"><a class="site-page" href="/publications/"> Publications</a></li><li class="menus_item"><a class="site-page" href="/project/"> Bachelor Thesis</a></li><li class="menus_item"><a class="site-page" href="/journal-club/"> Journal Club</a></li></ul></div></header><main id="page_main"><div class="side-card sticky"><div class="card-wrap" itemscope itemtype="http://schema.org/Person"><div class="author-avatar"><img class="avatar-img" src="/img/OIP.jpeg" onerror="this.onerror=null;this.src='/img/profile.png'" alt="avatar"></div><div class="author-discrip"><h3>Guo Jiayi</h3><p class="author-bio">A master student in Chemical Biology at University of Geneva and EPFL. Studying Machine Learning and Computional biology at this stage!</p></div><div class="author-links"><button class="btn m-social-links">Links</button><ul class="social-icons"><li><a class="social-icon" href="https://twitter.com/NicoNi114514/" target="_blank"><i class="fab fa-twitter" aria-hidden="true"></i></a></li><li><a class="social-icon" href="https://github.com/NicoNiCoN11/" target="_blank"><i class="fab fa-github" aria-hidden="true"></i></a></li><li><a class="social-icon" href="https://www.linkedin.com/in/jiayi-guo-a7b5aa297/" target="_blank"><i class="fab fa-linkedin" aria-hidden="true"></i></a></li><li><a class="social-icon" href="mailto:gjqtime@gmail.com" target="_blank"><i class="fas fa-envelope" aria-hidden="true"></i></a></li></ul><ul class="social-links"><li><a class="e-social-link" href="/" target="_blank"><i class="fas fa-graduation-cap" aria-hidden="true"></i><span>Google Scholar</span></a></li><li><a class="e-social-link" href="/" target="_blank"><i class="fab fa-orcid" aria-hidden="true"></i><span>ORCID</span></a></li></ul></div><a class="cv-links" href="/attaches/CV_2024.pdf" target="_blank"><i class="fas fa-file-pdf" aria-hidden="true"><span>Click here to see my CV.</span></i></a></div></div><div class="page" itemscope itemtype="http://schema.org/CreativeWork"><h2 class="page-title">Journal Club</h2><article><ul>
<li><strong>Click the cover to access my presentation slides</strong></li>
</ul>
<div class="papers">
    <h3>1. Multilayered omics reveal sex- and depot-dependent adipose progenitor cell heterogeneity</h3>
    <p>Rana K. Gupta, Yibo Wu</p>
    <p>Published in Cell Metabolism</p>
    <p>Key words: Proteomics analysis, RNA-seq analysis, Lineage Tracing analysis</p>
  <a href="/images/journal/pdf1.pdf" class="paper-item">
    <img src="/images/journal/paper1.jpg" alt="Cover" style="width: 500px;">
  </a>

<ul>
<li><strong>Comments:</strong> The authors of this paper use various methods to analyse the heterogeneity of adipose progenitor cells in mice, and conduct a research focusing on 3 different dimensions: Sex, Location, and Cell type, which largely clarifed the heterogeneity, and demonstrated the power of combining multiple omics in research.</li>
</ul>
<div class="papers">
    <h3>2. Menin “reads” H3K79me2 mark in anucleosomal context</h3>
    <p>Xiang David Li,Zhai Yuanliang</p>
    <p>Published in Science</p>
    <p>Key words: H3K73me2, Photocross-linking, Structral biology</p>
  <a href="/images/journal/pdf2.pdf" class="paper-item">
    <img src="/images/journal/paper2.jpg" alt="Cover" style="width: 500px;">
  </a>

<ul>
<li><strong>Comments:</strong> Identifying the “Reader” of H3K79me2 has puzzled the scientists for many years, but Xiang David Li and his collegues developed a Chemical probe method to identify this “Reader”, and the experiment design of this article is wonderful, they cleverly combined different analysis tools together, while they identified this “Reader”, they also described its working mechanism as detailed as possible.</li>
</ul>
<div class="papers">
    <h3>3. Control of osteoblast regeneration by a train of Erk activity waves</h3>
    <p>Stefano Di Talia, Kenneth D. Poss</p>
    <p>Published in Nature</p>
    <p>Key words: Regeneration, Erk signaling, Mathematic model</p>
  <a href="/images/journal/pdf3.pdf" class="paper-item">
    <img src="/images/journal/paper3.jpg" alt="Cover" style="width: 500px;">
  </a>

<ul>
<li><strong>Comments:</strong> This paper employs a mathematical model to simulate the regeneration process and successfully addresses the visualization of Erk signaling in regenerating tissues. Using this model and the visualizing tools, the researchers were able to determine how Erk spreads in zebrafish scales and uncover aspects of the underlying mechanisms of the regeneration process. Most importantly, these methods also offer valuable insights for similar studies in related fields.</li>
</ul>
<div class="papers">
    <h3>4. Cryo-EM structure determination of small therapeutic protein targets at 3 Å-resolution using a rigid imaging scaffold</h3>
    <p>Yeates, Todd O</p>
    <p>Published in PNAS</p>
    <p>Key words: Structural Biology, small protein imaging, Protein assemblies</p>
  <a href="/images/journal/pdf5.pdf" class="paper-item">
    <img src="/images/journal/Paper5.jpg" alt="Cover" style="width: 500px;">
  </a>

<ul>
<li><strong>Comments:</strong> Identifying the structure of small proteins with Crto-EM has confused scientists for a along time, this paper further improved the idea of making use of a bigger scaffold self-assembly protein to image the small protein, they rigidified the DARPin through modified the protrution direction of DARPin and making it less flexible. To validate its reliability, the author analyse the Cryo-EM structure of KRAS protein and get a good resolution. But the author do not mention any limitations of this technology, for example, the drug can not bind to where the DARPin binds to.</li>
</ul>
<div class="papers">
    <h3>5. Small molecule photocatalysis enables drug target identiﬁcation via energy transfer</h3>
    <p>David W. C. MacMillan</p>
    <p>Published in PNAS</p>
    <p>Key words: Small molecule catalysis, Target identiﬁcation, Signal amplification</p>
  <a href="/images/journal/pdf6.pdf" class="paper-item">
    <img src="/images/journal/paper6.jpg" alt="Cover" style="width: 500px;">
  </a>

<ul>
<li><strong>Comments:</strong></li>
</ul>
<div class="papers">
    <h3>6. Selection of DNA-encoded chemical libraries against endogenous membrane proteins on live cells</h3>
    <p>Li Xiaoyu, Li Yizhou</p>
    <p>Published in Nature Chemistry</p>
    <p>Key words: DEL, Ligand identiﬁcation, </p>
  <a href="/images/journal/pdf7.pdf" class="paper-item">
    <img src="/images/journal/paper7.jpg" alt="Cover" style="width: 500px;">
  </a>
</article></div></main><div class="nav-wrap"><div class="nav"><button class="site-nav"><div class="navicon"></div></button><ul class="nav_items"><li class="nav_item"><a class="nav-page" href="/about/"> About</a></li><li class="nav_item"><a class="nav-page" href="/publications/"> Publications</a></li><li class="nav_item"><a class="nav-page" href="/project/"> Bachelor Thesis</a></li><li class="nav_item"><a class="nav-page" href="/journal-club/"> Journal Club</a></li></ul></div><div class="cd-top"><i class="fa fa-arrow-up" aria-hidden="true"></i></div></div><footer id="page_footer"><div class="footer_wrap"><div class="copyright">&copy;2024 by Guo Jiayi</div></div></footer><script src="https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/jquery-pjax@latest/jquery.pjax.min.js"></script><script src="/js/main.js"></script></body></html>