<!DOCTYPE html><html lang="en"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=5"><title>GuoJiayi</title><meta name="author" content="Guo Jiayi"><link rel="shortcut icon" href="/img/favicon.png"><link rel="stylesheet" href="/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.13.0/css/all.min.css"><link rel="stylesheet" href="/css/custom.css"><meta name="generator" content="Hexo 7.3.0"></head><body><header id="page_header"><div class="header_wrap"><div id="blog_name"><a class="blog_title" id="site-name" href="/">GuoJiayi</a></div><button class="menus_icon"><div class="navicon"></div></button><ul class="menus_items"><li class="menus_item"><a class="site-page" href="/about/"> About</a></li><li class="menus_item"><a class="site-page" href="/journal-club/"> Journal Club</a></li><li class="menus_item"><a class="site-page" href="/publications/"> Publications</a></li><li class="menus_item"><a class="site-page" href="/project/"> Bachelor Thesis</a></li><li class="menus_item"><a class="site-page" href="/blogs/"> Blogs</a></li></ul></div></header><main id="page_main"><div class="side-card sticky"><div class="card-wrap" itemscope itemtype="http://schema.org/Person"><div class="author-avatar"><img class="avatar-img" src="/img/OIP.jpeg" onerror="this.onerror=null;this.src='/img/profile.png'" alt="avatar"></div><div class="author-discrip"><h3>Guo Jiayi</h3><p class="author-bio">A master student in Chemical Biology at University of Geneva and EPFL. Studying Machine Learning and Computional biology at this stage!</p></div><div class="author-links"><button class="btn m-social-links">Links</button><ul class="social-icons"><li><a class="social-icon" href="https://twitter.com/NicoNi114514/" target="_blank"><i class="fab fa-twitter" aria-hidden="true"></i></a></li><li><a class="social-icon" href="https://github.com/NicoNiCoN11/" target="_blank"><i class="fab fa-github" aria-hidden="true"></i></a></li><li><a class="social-icon" href="https://www.linkedin.com/in/jiayi-guo-a7b5aa297/" target="_blank"><i class="fab fa-linkedin" aria-hidden="true"></i></a></li><li><a class="social-icon" href="mailto:gjqtime@gmail.com" target="_blank"><i class="fas fa-envelope" aria-hidden="true"></i></a></li></ul><ul class="social-links"><li><a class="e-social-link" href="/" target="_blank"><i class="fas fa-graduation-cap" aria-hidden="true"></i><span>Google Scholar</span></a></li><li><a class="e-social-link" href="/" target="_blank"><i class="fab fa-orcid" aria-hidden="true"></i><span>ORCID</span></a></li></ul></div><a class="cv-links" href="/attaches/CV_2024.pdf" target="_blank"><i class="fas fa-file-pdf" aria-hidden="true"><span>Click here to see my CV.</span></i></a></div></div><div class="page" itemscope itemtype="http://schema.org/CreativeWork"><h2 class="page-title">Bachelor Thesis</h2><article><h3 id="Benchmarking-Algorithms-for-Batch-Correcttion-in-Single-cell-ATAC-seq-data-Analysis"><a href="#Benchmarking-Algorithms-for-Batch-Correcttion-in-Single-cell-ATAC-seq-data-Analysis" class="headerlink" title="Benchmarking Algorithms for Batch Correcttion  in Single-cell ATAC-seq data Analysis"></a>Benchmarking Algorithms for Batch Correcttion  in Single-cell ATAC-seq data Analysis</h3><ul>
<li><strong>Workflow</strong>:<img src="/images/Project3.jpg" alt="Research Project Image" title="View of Research Project"></li>
<li><strong>Abstract</strong>:【Purpose】 With the development of high-throughput sequencing technology and single-cell techniques, single-cell chromatin accessibility analysis (Single cell ATAC-seq) has gradually become one of the main means to study the regulatory networks of genome expression. However, in most Single cell ATAC analyses, one of the core issues is how to eliminate unnecessary batch effects caused by different experimental batches. This study aims to compare four batch effect correction algorithms (MNN, Harmony, BBKNN, scVI) in handling two different single-cell ATAC-seq data analysis tasks, to evaluate their performance in preserving biological features and removing batch effects, and to conduct preliminary benchmark testing.【Content】This study first downloads fastqc files of 7 sets of brain single-cell ATAC-seq data published online from the GEO database, and preprocesses the data using software such as Trimgalore, BWA-MEM, SAMTOOL, etc. The obtained Anndata data is then divided into two datasets based on existing batch effects, and four algorithms are used to respectively eliminate batch effects caused by different donors and those caused by different donors and sampling sites. Finally, four metrics are used to evaluate the performance of different algorithms and their preservation effects on biological features, providing references for algorithm selection.【Conclusion】When dealing with datasets with a single source of batch effects, Harmony usually performs excellently in all metrics. When handling datasets with complex batch effects sources, BBKNN performs better. The Harmony and MNN algorithms based on the linear embedding model principle perform outstandingly in different datasets. BBKNN is notably inferior to other algorithms in handling mixed samples. Additionally, the deep learning algorithm scVI also shows stable performance. Through such systematic evaluation and analysis, this study provides a deeper understanding and effective solutions for handling batch effects in single-cell chromatin accessibility data in the field of biomedicine, promoting the development and application of related research, and further revealing the application prospects of deep learning principles in single-cell dataset analysis.</li>
<li><strong>Key words</strong>: Single cell ATAC-seq; Batch effect remval; Benchmark; Single cell data analysis</li>
<li><strong>link</strong>: <a href="/images/Bachelor_thesis.pdf">Read the paper(In Chinese)</a></li>
</ul>
</article></div></main><div class="nav-wrap"><div class="nav"><button class="site-nav"><div class="navicon"></div></button><ul class="nav_items"><li class="nav_item"><a class="nav-page" href="/about/"> About</a></li><li class="nav_item"><a class="nav-page" href="/journal-club/"> Journal Club</a></li><li class="nav_item"><a class="nav-page" href="/publications/"> Publications</a></li><li class="nav_item"><a class="nav-page" href="/project/"> Bachelor Thesis</a></li><li class="nav_item"><a class="nav-page" href="/blogs/"> Blogs</a></li></ul></div><div class="cd-top"><i class="fa fa-arrow-up" aria-hidden="true"></i></div></div><footer id="page_footer"><div class="footer_wrap"><div class="copyright">&copy;2024 by Guo Jiayi</div></div></footer><script src="https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/jquery-pjax@latest/jquery.pjax.min.js"></script><script src="/js/main.js"></script></body></html>