<!DOCTYPE html><html lang="en"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=5"><title>GuoJiayi</title><meta name="author" content="Guo Jiayi"><link rel="shortcut icon" href="/img/favicon.png"><link rel="stylesheet" href="/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.13.0/css/all.min.css"><link rel="stylesheet" href="/css/custom.css"><meta name="generator" content="Hexo 7.3.0"></head><body><header id="page_header"><div class="header_wrap"><div id="blog_name"><a class="blog_title" id="site-name" href="/">GuoJiayi</a></div><button class="menus_icon"><div class="navicon"></div></button><ul class="menus_items"><li class="menus_item"><a class="site-page" href="/about/"> About</a></li><li class="menus_item"><a class="site-page" href="/journal-club/"> Journal Club</a></li><li class="menus_item"><a class="site-page" href="/publications/"> Publications</a></li><li class="menus_item"><a class="site-page" href="/project/"> Bachelor Thesis</a></li><li class="menus_item"><a class="site-page" href="/blogs/"> Blogs</a></li></ul></div></header><main id="page_main"><div class="side-card sticky"><div class="card-wrap" itemscope itemtype="http://schema.org/Person"><div class="author-avatar"><img class="avatar-img" src="/img/OIP.jpeg" onerror="this.onerror=null;this.src='/img/profile.png'" alt="avatar"></div><div class="author-discrip"><h3>Guo Jiayi</h3><p class="author-bio">A master student in Chemical Biology at University of Geneva and EPFL. Studying Machine Learning and Computional biology at this stage!</p></div><div class="author-links"><button class="btn m-social-links">Links</button><ul class="social-icons"><li><a class="social-icon" href="https://twitter.com/NicoNi114514/" target="_blank"><i class="fab fa-twitter" aria-hidden="true"></i></a></li><li><a class="social-icon" href="https://github.com/NicoNiCoN11/" target="_blank"><i class="fab fa-github" aria-hidden="true"></i></a></li><li><a class="social-icon" href="https://www.linkedin.com/in/jiayi-guo-a7b5aa297/" target="_blank"><i class="fab fa-linkedin" aria-hidden="true"></i></a></li><li><a class="social-icon" href="mailto:gjqtime@gmail.com" target="_blank"><i class="fas fa-envelope" aria-hidden="true"></i></a></li></ul><ul class="social-links"><li><a class="e-social-link" href="/" target="_blank"><i class="fas fa-graduation-cap" aria-hidden="true"></i><span>Google Scholar</span></a></li><li><a class="e-social-link" href="/" target="_blank"><i class="fab fa-orcid" aria-hidden="true"></i><span>ORCID</span></a></li></ul></div><a class="cv-links" href="/attaches/CV_2024.pdf" target="_blank"><i class="fas fa-file-pdf" aria-hidden="true"><span>Click here to see my CV.</span></i></a></div></div><div class="page" itemscope itemtype="http://schema.org/CreativeWork"><h2 class="page-title">A Brief Introduction to Computational Biology</h2><p class="meta">Author: Jiayi Guo<br>Date: January 27, 2025</p><article><h2 id="Introduction"><a href="#Introduction" class="headerlink" title="Introduction"></a>Introduction</h2><p>Computational biology is an interdisciplinary field that connects computer science, biology, and statistics, data science, which has a relatively big overlap with Bioinformatics.</p>
<p>In this blog, I will introduce some Bioinformatics tools and databases as well as some basic concepts and algorithms of computational biology. This will help beginners better understand how computational biology works and how to design a computational biology research project or compensate your research with some Bioinformatics analysis. The main content of this instruction refers to the “Elements of Bioinformatics” courses in the autumn of 2024 at Geneva and <a target="_blank" rel="noopener" href="https://www.bioinformaticsalgorithms.org/">“Bioinformatics Algorithms: An Active Learning Approach”</a> published by Phillip Compeau and Pavel Pevzner in 2015.</p>
<hr>
<h2 id="Outline"><a href="#Outline" class="headerlink" title="Outline"></a>Outline</h2><ol start="0">
<li>Preface</li>
<li>Main Biological Databases<br>1.0 Servers<br>1.1 Sequence database(Protein, DNA, RNA…)<br>1.2 Genomic Database<br>1.3 Protein structure databases<br>1.4 Other biological databases</li>
<li>Bioinformatics and Computational Biological Analysis Methods<br> 2.1 GeneOntology and its associated analysis methods<br> 2.2 Gene anotation: recognizing the structure of gene automatically<br> 2.3 From pattern to domain: identify protein consensus sequences<br> 2.4 Homologs identification: a method to review the relationship between proteins<br> 2.5 Protein Structure Prediction: from the past to future<br> 2.6 Constructing a phylogeny trees with NG Phylogeny </li>
<li>Some basic algorithm of Computational Biology<br>3.1 BLAST<br>3.2 Pair-wise alignment<br>3.3 Multiple sequence alignment(MSA)</li>
<li>My understanding of system biology</li>
</ol>
<hr>
<h3 id="0-Preface-This-section-does-not-refer-to-any-sources"><a href="#0-Preface-This-section-does-not-refer-to-any-sources" class="headerlink" title="0. Preface(This section does not refer to any sources)"></a>0. Preface(This section does not refer to any sources)</h3><p>Before the main part, I’d like to clarify the definition of the two terms: Bioinformatics and Computational Biology. Since befofe we learn something, it would be better that we can have a whole picture of what you are going to learn.<br>Bioinformatics is an interdisciplinary field that applies computational methodologies for managing and interpreting large-scale biological data. A typical bioinformatics study involves <strong>four key steps</strong>: </p>
<ol>
<li><strong>Data</strong> Acquisition</li>
<li><strong>Data</strong> Storage</li>
<li><strong>Data</strong> Distribution</li>
<li><strong>Data</strong> Analysis</li>
</ol>
<p>From these four elements, you must notice that bioinformatics extremely focouses on the word “<strong>data</strong>“, which is the core of bioinformatics.<br>Generally speaking Bioinformatics often focuses on building and using software tools and databases, It not so much a systematic discipline as a collection of utilities to address biological problems from the data collected from different database or experiments. Thus, an outstanding bioinformaticians must have a strong ability of data analysing and data mining.</p>
<p>In contrast, computational biology is a systematic discipline that merges biology, computer science, mathematics, and statistics to tackle a wide range of biological questions. The main elements involved in computational biology is:</p>
<ol>
<li>Simulating Biological process with Methematical models</li>
<li>Predicting information with Statistical Learning methods</li>
<li>Designing or Improving Bioinfo-tools using computational algorithms</li>
</ol>
<p>Computational biology is a <strong>high-level</strong> interdiscipline(there’s no “Computational Biology” major in undergraduate study) which emphasizes intergrating algorithms and the development of theoretical models to address specific biological problems. It incorporates various methodological and conceptual approaches—such as machine learning, mathematical modeling, and simulation—to explore complex biological systems in a more quantitative and predictive manner.<br>It’s important to tell the differences between Computational Biology and Bioinformatics, since the work they are doing is quite different, Bioinformaticians are basicly data analysts dealing with biological data, which is what we called “Dry lab work”, while Computational Biologists are still biologist who are studying without tubes and various lab instruments, but with Computers and different models as we mentioned above.<br>Knowing the difference between these two terms, you also need to realize how far you want to achieve based on our goal.<br>If you want to be a Bioinfor Data analyst, it would be enough for you to get familar with different databases, bioinformatics tools and at least one coding language(Python or R) for analysing data and writing scripts (I assume you already known the basis of Moleular Biology and Biochemistry). A good bioinformatician can be trained by performing data analysing tasks to accumulate experiences and improve the efficiency.<br>In contrast, if your goal is to be a computational biologist (which is also my goal) who can perform research to answer scientific questions without microscopes and enzymes but with computers everywhere, it’s both necessary to learn the fundmentals of Biology and computer science, find and address problems from the perspective of both discilines. These abilities(scientific thinking, problem addressing, and interdisciplinary views…) is usually be trained when actually dive into an science environment, and the journey of computational biologist only start after or during finishing a PhD degree.</p>
<hr>
<h2 id="1-Main-Biological-Databases"><a href="#1-Main-Biological-Databases" class="headerlink" title="1. Main Biological Databases"></a>1. Main Biological Databases</h2><p>Databases are usually a collection of structured and searchable data, and these data are also updated periodically, different databases often have crosslink which can link to other database. They are usually very powerful tools for data mining or finding the reference of your research.</p>
<h3 id="1-0-Servers"><a href="#1-0-Servers" class="headerlink" title="1.0 Servers"></a>1.0 Servers</h3><p>A webserver is a computer(from a given insititute) that provide services(databases and tools) to other computers.<br>There are 5 main biological servers:</p>
<ul>
<li><a href="www.ncbi.nlm.nih.gov/">NCBI</a></li>
<li><a href="www.uniprot.org">UniProt</a> (Maintained by SIB, EBI and PIR)</li>
<li><a href="www.expasy.org/">ExPASy</a>(Maintained by SIB)</li>
<li><a href="www.ebi.ac.uk/">EBI</a></li>
<li><a href="www.genome.jp">JapaneseGenomeNet</a></li>
</ul>
<h3 id="1-1-Sequence-databases"><a href="#1-1-Sequence-databases" class="headerlink" title="1.1 Sequence databases"></a>1.1 Sequence databases</h3><p>Regarding the sequence databases, in biology, it usually refers to DNA, RNA and Protein sequence data. As shown in <strong>Figure 1</strong>, for DNA, RNA sequences data, <strong><a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/ena/browser/home">ENA</a>, <a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/genbank/">GenBank</a> and <a target="_blank" rel="noopener" href="https://www.ddbj.nig.ac.jp/index-e.html">DDBJ</a></strong> (Primary archives) are usually the major databases accepting sequence submission from researchers and these data usually comes from direct sequencing.</p>
<style>
  .fig-caption {
    font-size: 12px;
    color: gray;
  }
</style>
<div style="text-align: center;">
  <img src="/images/blogs/Blog1/image1.png" alt="Fig1" style="width: 550px; height: auto;">
  <p class="fig-caption">Fig 1. Origin of Sequence</p>
</div>

<p>An important feature of these primary archives is highly redundant(which means nothing goes out) and their annotations are not updating, so there are many errors in both sequences and annotations.<br><strong><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/refseq/">Refseq</a>, <a target="_blank" rel="noopener" href="https://www.ensembl.org/index.html">Ensembl</a></strong> (Figure 2)take sequences from primary archives, and apply quality control and gene annotation pipeline to provide the “reference” annotations.</p>
<style>
  .fig-caption {
    font-size: 12px;
    color: gray;
  }
</style>
<div style="text-align: center;">
  <img src="/images/blogs/Blog1/Nadatabase.png" alt="Fig2" style="width: 400px; height: auto;">
  <p class="fig-caption">Fig 2. Ensembl and Refseq</p></div>

<p>In Refseq and Ensembl, we can also find the protein sequences derived from gene prediction, which are not submitted to the primary databases.</p>
<blockquote>
<p><strong>Note</strong>: Because of the different methods used for gene prediciton, the final number of proteins can also be different.</p>
</blockquote>
<p>For protein sequences, a relatively large persentage of them comes from translations of the DNA sequences or predicting the gene structure and then deduce its products(about99%), only 1% come from direct protein sequencing.<br>Mostly used protein databese including <strong><a target="_blank" rel="noopener" href="https://www.uniprot.org/">UniProtKB</a></strong> and <strong><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/protein">NCBI protein</a></strong>, both of them are intergrated databases, for UniProtKB, it intergrated data from PDB, PIR, PRF into it database. One of its sub database Swiss-Prot only contains canonical sequences, it usually has these features:</p>
<ul>
<li>One gene, one UniProtKB&#x2F;Swiss-Prot entry.</li>
<li>AlphaFold usually use the canonical sequences.</li>
<li>Has annotations from manuel or experimental evidences.</li>
</ul>
<p>TrEMBL is also a sub database of UniProtKB, but this is exactly the opposite of the Swiss-Prot, every entry of it <strong>corresponds to a single protein, instead of a gene</strong>, and its annotations are mostly generated automatically from diverse sources like Interpro predictions.<br>Their differences has been shown in the following Figure 3.</p>
<style>
  .fig-caption {
    font-size: 12px;
    color: gray;
  }
</style>
<div style="text-align: center;">
  <img src="/images/blogs/Blog1/difference.png" alt="Fig3" style="width: 500px; height: auto;">
  <p class="fig-caption">Fig 3. Differences between Swiss-Prot and TrEMBL</p></div>

<p>In contrast NCBI protein database did not intergrate PDB, PIR and PRF, instead, it provide these information sepratedly.<br>As the description and interface shown in Figure 4. NCBI protein database is not only have access to different data, it also provide access to some tools.</p>
<style>
  .fig-caption {
    font-size: 12px;
    color: gray;
  }
</style>
<div style="text-align: center;">
  <img src="/images/blogs/Blog1/ncbipro.png" alt="Fig4" style="width: 550px; height: auto;">
  <p class="fig-caption">Fig 4. NCBI protein description</p></div>

<h3 id="1-2-Genomic-Database"><a href="#1-2-Genomic-Database" class="headerlink" title="1.2 Genomic Database"></a>1.2 Genomic Database</h3><p>Genomic databases are also called <strong>Model Organism Database(MOD)</strong>, these database usually provide information on Genes, Chromosomal Location, Gene Expression, Mutations&amp;Phenotypes and Homology of the model organisms, as shown in Figure 5, in addition to well-known mice, other model oragnisms including yeast, fruit fly, etc.</p>
<style>
  .fig-caption {
    font-size: 12px;
    color: gray;
  }
</style>
<div style="text-align: center;">
  <img src="/images/blogs/Blog1/model_or.jpg" alt="Fig5" style="width: 450px; height: auto;">
  <p class="fig-caption">Fig 5. Model Oragnism</p></div>

<p>For most of the model organisms there exist an MOD of it, for example, we have <a target="_blank" rel="noopener" href="https://www.omim.org/">OMIM(human)</a>, <a target="_blank" rel="noopener" href="https://www.informatics.jax.org/">MGI(mouse)</a>, <a target="_blank" rel="noopener" href="https://flybase.org/">Flybase(Drosophila)</a>,<a target="_blank" rel="noopener" href="https://www.yeastgenome.org/">SGD(yeast)</a>, <a target="_blank" rel="noopener" href="https://maizegdb.org/">MaizeGDB(maize)</a>, <a target="_blank" rel="noopener" href="http://genolist.pasteur.fr/SubtiList/">SubtiList(B.subtilis)</a>, <a target="_blank" rel="noopener" href="https://zfin.org/">ZFIN(zebrafish)</a>,<a target="_blank" rel="noopener" href="https://wormbase.org/#012-34-5">Wormbase(C.elegan)</a>, <a target="_blank" rel="noopener" href="https://www.xenbase.org/xenbase/">Xenbase(frog)</a>.<br>These database typically include:</p>
<ul>
<li><ol>
<li>Genome data: Genome sequence(only link to sequence database), gene annotations and variants.</li>
</ol>
</li>
<li><ol start="2">
<li>Phenotypic data: Traits associated with mutations.</li>
</ol>
</li>
<li><ol start="3">
<li>Functional data: Gene expression patterns, protein interactions as well as pathways</li>
</ol>
</li>
<li><ol start="4">
<li>Cross-reference links: Connects to relevant publications or databases</li>
</ol>
</li>
<li><ol start="5">
<li>Experimental tool: Some built-in tools for sequence alignment or genome browsing.</li>
</ol>
</li>
</ul>
<p>Here, we mainly introduce the human genomic database <a target="_blank" rel="noopener" href="https://www.omim.org/">OMIM</a>(Online Mendelian Inheritance in Man), in OMIM, there are in total 20300 genes in which 16800 of them has known biological function from UniProtKB, it focuses on the relationship between phenotype and genotype.<br>In OMIM database, the accesstion with a prefix “*” is a gene entry, in which we can find its associated diseases and other descriptions as shown in <a target="_blank" rel="noopener" href="https://omim.org/entry/164870?search=HER2&highlight=her2#geneMap">Figure 6</a>.</p>
<style>
  .fig-caption {
    font-size: 12px;
    color: gray;
  }
</style>
<div style="text-align: center;">
  <img src="/images/blogs/Blog1/gene_entry.png" alt="Fig6" style="width: 550px; height: auto;">
  <p class="fig-caption">Fig 6. Example gene entry</p></div>

<p>For disease entry, the prefix will be “#”, as shown in <a target="_blank" rel="noopener" href="https://omim.org/entry/607822?search=Alzheimer%27s%20disease&highlight=alzheimer,disease">Figure 7</a>, the entry of disease usually gives the access to its clinical synopsis and its genetic relationships.</p>
<style>
    .fig-caption {
    font-size: 12px;
    color: gray;
  }
</style>
<div style="text-align: center;">
  <img src="/images/blogs/Blog1/disease_entry.png" alt="Fig7" style="width: 550px; height: auto;">
  <p class="fig-caption">Fig 7. Example disease entry</p></div>

<blockquote>
<p><strong>Note</strong>: Most of the identified associated genes are “Markers” instead of actual causative variants.</p>
</blockquote>
<p>An important tool for searching genome or reading genome information is Genome Browser, which allow us visualizing data(genome, SNPs, mRNAs) for a given gene. There are three most popular genome browser most often used:</p>
<ul>
<li>1 <a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/gdv/">GDV(NCBI)</a></li>
<li>2 <a target="_blank" rel="noopener" href="https://genome.ucsc.edu/">UCSC genome browser</a></li>
<li>3 <a target="_blank" rel="noopener" href="https://www.ensembl.org/Homo_sapiens/Location/Genome">Ensembl</a></li>
</ul>
<h3 id="1-3-Protein-Structure-Database"><a href="#1-3-Protein-Structure-Database" class="headerlink" title="1.3 Protein Structure Database"></a>1.3 Protein Structure Database</h3><p>There is only one major protein structure database which is <a target="_blank" rel="noopener" href="https://www.rcsb.org/">Protein Data Bank(PDB)</a>, but it can be accessed through different servers: <a href="www.rcsb.org/pdb/">RCSB</a>, <a href="www.ebi.ac.uk/pdbe/">EBI</a>, <a href="www.pdbj.org">PDBJ</a>.<br>In PDB database, it currently(till 2025.1.31) has 230000 protein structure data for 47000 proteins which is highly redundant. PDB database storing data in .pdb format, a .pdb file ussally looks like the Figure 8, it shows the spatial coordinates of each atom of each amino acid:</p>
<div style="text-align: center;">
  <img src="/images/blogs/Blog1/pdb.png" alt="Fig8" style="width: 550px; height: auto;">
  <p class="fig-caption">Fig 8. .pdb file example</p></div>

<p>We can use some associated Softwares to visualize the coresponding 3D structures(<a target="_blank" rel="noopener" href="https://spdbv.unil.ch/disclaim.html#">SwissPDB-viewer</a>,<a target="_blank" rel="noopener" href="https://www.cgl.ucsf.edu/chimerax/download.html">ChimeraX</a>, <a target="_blank" rel="noopener" href="http://rasmol.org/">Rasmol</a>), we can also access the structure information via UniProtKB, here is an <a target="_blank" rel="noopener" href="https://www.uniprot.org/uniprotkb/P04150/entry#structure">quick example</a>.<br>As the development of Machine Learning models, more specificly, the Neuron Network, DeepMind developed an predictive model named <a target="_blank" rel="noopener" href="https://alphafold.com/">AlphaFold</a> which set off a wave in the field of computational biology. AlphaFold can receive a sequence as its input and predict its corresponding structures with the trained deep-learning model, it now serves as a powerful tools in a lot of different research fields like protein design or target identifying.<br>Another predicting model is <a target="_blank" rel="noopener" href="https://swissmodel.expasy.org/interactive">SWISS-MODEL</a>, which was developed by SIB, but it use the method of comparative modeling, algning the input sequence to known PDB structure, although it also has a high accuracy, it’s difficult to find a high quality template for the query sequence to align with.</p>
<h3 id="1-4Other-Databases"><a href="#1-4Other-Databases" class="headerlink" title="1.4Other Databases"></a>1.4Other Databases</h3><p>Apart from sequence databases and 3D structure databases, other biological databases also exist, if we open NCBI we can see a list as shown in Figure 9, except proteins, RNAs and DNAs, we can also access the database for SNPs, which allow us to find human specific human specific variations and its frequency and positions. Also in NCBI genome database, we can find the reference genome of various species.</p>
<blockquote>
<p><strong>SNP (Single Neucleotide Polymorphisms)</strong> are some unique genetic differences between individuals</p>
</blockquote>
<div style="text-align: center;">
  <img src="/images/blogs/Blog1/NCBIlist.png" alt="Fig9" style="width: 650px; height: auto;">
  <p class="fig-caption">Fig 9. NCBI database list</p></div>

<p>We now have databases for sequences, proteins structures, and even mutations. What else do we need?<br>Exactly, we need information for certain metabolism pathways or enzyme pathways<br>Regarding this need, we have <a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/chebi/">ChEBI</a> which focused on small chemicals and describe their biological metabolism pathways. Here is a <a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/chebi/searchId.do?chebiId=CHEBI:27740">quick example</a> for what is shown in its entry. In ChEBI, there is also a different term called ChEBI ontology, which describes the hierarchy of its metabolism. We can use this ChEBI ontology to search all proteins involved in certain chemicals’ metabolism pathways in certain organisms, here is <a target="_blank" rel="noopener" href="https://www.uniprot.org/uniprotkb?query=(chebi:%22CHEBI:17716%22)">an example</a> shows the protein involved in lactose metabolism across organisms.<br>Another similar database as ChEBI is <a target="_blank" rel="noopener" href="https://pubchem.ncbi.nlm.nih.gov/">PubChem</a> maintained by NCBI.<br>For enzyme reactions, there is <a target="_blank" rel="noopener" href="https://www.rhea-db.org/">Rhea</a>, which has a <strong>non-redundant</strong> resource of biochemical reactions details(participants, cross-reference) of enzymes <strong>instead of enzyme itself</strong>. In a word, in Rhea, each entry represent a biochemical reaction.</p>
<blockquote>
<p>Rhea is the standard enzyme annotation of UniProtKB database. </p>
</blockquote>
<p>As shown in <a target="_blank" rel="noopener" href="https://www.rhea-db.org/rhea/11844">Figure 10</a>, we can find the ChEBI Ontology accesstion in Rhea to find the proteins related to the chemicals produced in this biochemical reaction with UniProtKB. It can be a powerful tool to conduct some data mining.</p>
<div style="text-align: center;">
  <img src="/images/blogs/Blog1/Rhea.png" alt="Fig10" style="width: 550px; height: auto;">
  <p class="fig-caption">Fig 10. NCBI database list</p></div>

<p>As for Pathway databases, there are <a target="_blank" rel="noopener" href="https://www.genome.jp/kegg/pathway.html">KEGG</a> (manually drawn pathway maps) and <a target="_blank" rel="noopener" href="https://metacyc.org/">METACYC</a>(experimentally vertified database) where we can find the involved pathways of Genes or Proteins. An using example for them is GULO gene, in KEGG, we can compare pathway across different species, and in METACYC, we can search for a single gene or protein to see how many pathways it’s been involved in.</p>
<p>With the development of <strong>System Biology</strong>, the interactome database which contains the information about Protein-Protein interaction, in UniProtKB databases we can also find the cross-references to the interactome database like IntAct, STRING as shown in Figure 11.</p>
<div style="text-align: center;">
  <img src="/images/blogs/Blog1/PPI.png" alt="Fig11" style="width: 500px; height: auto;">
  <p class="fig-caption">Fig 11. Cross-reference to PPI databases</p></div>

<p>As mentioned, <a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/intact/home">IntAct</a> is one of the main interactome database, whose data comes form well-verified publications or direct submissions. It can give a PPI network of genes or proteins as shown in <a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/intact/search?query=GULO">Figure 12</a>.</p>
<div style="text-align: center;">
  <img src="/images/blogs/Blog1/IntAct.png" alt="Fig12" style="width: 500px; height: auto;">
  <p class="fig-caption">Fig 12. Cross-reference to PPI databases</p></div>

<p><a target="_blank" rel="noopener" href="https://string-db.org/">STRING</a> is also a database focusing on interactions between proteins. It also provide poweful analysis tools like GO enrichment analysis and clustering to mine more information from a given Protein-Protein interaction network. here is an <a target="_blank" rel="noopener" href="https://string-db.org/cgi/network?taskId=brJzJdzSPKSi&sessionId=biW9N0a2JYqB">quick search example</a>.</p>
<blockquote>
<p><strong>GO Enrichment Analysis(GOEA)</strong> is a method to test if certain GO term is enriched in a gene list. Which is a powerful method to see if a given gene list are involved in certein biological phenomenon.</p>
</blockquote>
<p>Another important database is <a target="_blank" rel="noopener" href="https://pubmed.ncbi.nlm.nih.gov/"><strong>PubMed</strong></a> I believe it is already a well-known literature database where you can find the literature about the topics you are interested in. So I’d like not to spend too much time on introducing this database, if you need any assistance you can read the official <a target="_blank" rel="noopener" href="https://pubmed.ncbi.nlm.nih.gov/help/">documentation of PubMed</a> or read this <a target="_blank" rel="noopener" href="https://paperpile.com/g/pubmed-guide/">Guide</a> from paperpile.</p>
<p>After introducing so many different biological databases, I believe you already know how to search for biological information of interest, before you start to learn Bioinformatics or Computational Biology, you might underestimate the importance of databases, thinking they are just a collection of data, but you must need to know that Bioinfomatics is basically a branch of data science, in most of the researches, we usually start from data mining, which can give us inspiration of finding out important biological information.<br>Nowadays, as Machine Learning and Deep Learning has become a trend in Computational Biology, databases also play a key role in providing the training data and Calculating the homologs or the domains of proteins.</p>
<p>In the Next chapter, I’d like to talk more about the methods people using in bioinformatics or computational biology.</p>
<hr>
<h2 id="2-Bioinformatics-and-Computational-Biological-Analysis-Methods"><a href="#2-Bioinformatics-and-Computational-Biological-Analysis-Methods" class="headerlink" title="2. Bioinformatics and Computational Biological Analysis Methods"></a>2. Bioinformatics and Computational Biological Analysis Methods</h2><p>Knowing different databases is far from designing a project by yourself, we still need to know some clasic methods used in the bioinformatics study. In this part, I will focous on the method of mining information or make predictions based on existing data.</p>
<h3 id="2-1-Gene-Ontology-and-its-associated-analysis-methods"><a href="#2-1-Gene-Ontology-and-its-associated-analysis-methods" class="headerlink" title="2.1 Gene Ontology and its associated analysis methods"></a>2.1 Gene Ontology and its associated analysis methods</h3><p>In the time between 1995 and 2002, which was the gloden age of Genome sequencing, during that time, a lot of new gene databases was created, and this also generated a problem that we can not only use the nomenclauture to annotate the gene’s functions or other characteristics.<br>And someone come up with a idea to design an ontology for genes at this time point.</p>
<blockquote>
<p><strong>Ontology</strong> is a set of concepts and categories in a subject that shows their properties and relationship between them.</p>
</blockquote>
<p>Ontology usually has two elements: <strong>Classses(nodes)</strong> and <strong>Relations(edges)</strong>, as shown in Figure 1, this is the formal representation of the knowledge in a domain of interest. Figure 1 means “Class ‘C1’ is related to Class’C2’ by the relation ‘R1’ “.</p>
<style>
    .fig-caption {
    font-size: 12px;
    color: gray;
  }
</style>
<div style="text-align: center;">
  <img src="/images/blogs/Blog1/Ontology_ele.png" alt="Fig1" style="width: 400px; height: auto;">
  <p class="fig-caption">Fig 1. Element in ontology</p></div>

<p>This stucture makes ontology usually have a hierarchical relationship among its classes, for example, “neuron” is the subclass(child) of “cell”, and “cell” is the superclass(parent) of “neuron” and each node(class) can have multiple parents or children and with different relations, it can represent a large amount of biological function annotations and can also make them organised. For example, we can make an simple ontology structure like Figure 2</p>
<blockquote>
<p><strong>Note</strong>: All the relations are <strong>inherited</strong>, for example, “Rabbit” is also an “organism” and a “vertebrate”</p>
</blockquote>
<div style="text-align: center;">
  <img src="/images/blogs/Blog1/Ontology_ex.png" alt="Fig2" style="width: 400px; height: auto;">
  <p class="fig-caption">Fig 2. Element in ontology</p></div>

<p>Now, we have an idea about what is ontology, now it’s easy to understand what is Gene Ontology, GO is just using an ontology structure to describe the biological information of a given gene. For Gene Ontology, we usually have 3 aspects which serverd as the three top class, they are <strong>Molecular Function</strong>; <strong>Biological process</strong>; <strong>Cellular Component</strong>, and the distribution of their subclasses inside these three aspects are uneven. The reason why GO is so widely used not only is it’s strict classification and hieriarchical strutue, it also provide a logic framework to allow computational tools to analyse large datasets and mine more information from a given information.<br>There are 3 types of GO annotation. They are:</p>
<ul>
<li><ol>
<li>A <strong>statement</strong> that links a gene product and a GO term</li>
</ol>
</li>
<li><ol start="2">
<li>A <strong>description</strong> of the evidence supporting the statement</li>
</ol>
</li>
<li><ol start="3">
<li>A <strong>reference</strong> citing the source of the information</li>
</ol>
</li>
</ul>
<p>So a GO annotation of a gene or gene product is usually based on a GO term or some kind of evidence.<br>These sources make GO a extremely powerful tool for gene function prediction: based on the given GO model of the known gene or gene product and their similarity&#x2F;homology with the unknown-function gene, we can infer the function of these unknown genes rapidly. Figure 3 shows a work flow to make gene annotations based on a evolutionary relationship(The PANTHER tree is a kind of phylogenetic trees)</p>
<div style="text-align: center;">
  <img src="/images/blogs/Blog1/Pango.png" alt="Fig3" style="width: 550px; height: auto;">
  <p class="fig-caption">Fig 3. Make gene annotation with Phylogenict way</p></div>

<p>Apart from manually making annotation, there are also methods that can make annotations automatically. There are over one billion GO annotations are made by automatic methods, for example, the UniProt annotation features like keyword, enzyme pathways can be automatically mapped as GO annotations(Figure 4), as well as domain informations</p>
<blockquote>
<p>Since domain are functional modules of protein, some domain can sometime imply some GO terms</p>
</blockquote>
<div style="text-align: center;">
  <img src="/images/blogs/Blog1/GO_auto.png" alt="Fig3" style="width: 500px; height: auto;">
  <p class="fig-caption">Fig 4. Keyword and Enzyme pathways in UniProt</p></div>

<p>If we want to access GO data, There are so many different ways, like NCBI, UniProt, we can usually find their GO terms when opening their entries. Also, we can use tools like <a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/QuickGO/">QuickGO</a> or <a target="_blank" rel="noopener" href="https://amigo.geneontology.org/amigo/landing">AmiGO</a> which are more dedicated tools for searching GO terms and making associated analysis.<br>There is an important analysis method called <strong>Gene Set Enrichment Analysis(GSEA)</strong>(also called functional enrichment analysis or pathway enrichment analysis) which is a method to identify classes of genes or proteins that are over-represented(or enriched) in a large set of genes or proteins(From <a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/Gene_set_enrichment_analysis">WikiPEDIA</a>), for GO terms Over-Represent Analysis, we can just use <a target="_blank" rel="noopener" href="https://davidbioinformatics.nih.gov/home.jsp">DAVID</a>(<strong>D</strong>atabse for <strong>A</strong>nnotation, <strong>V</strong>isualization and <strong>I</strong>ntegrated <strong>D</strong>iscovery). Here is an <a target="_blank" rel="noopener" href="https://davidbioinformatics.nih.gov/summary.jsp">example</a> of the result from DAVID analysis pipeline. Generally speaking, what DAVID does is analysing the gene list you entered as input and the species you choose as background, then it will calculate the counts of genes enriched at specific GO terms and calculate it’s significance (i.e. the p-value) to illustrate the<br>$p &#x3D; \frac{\binom{A+B}{A} \binom{C+D}{C}}{\binom{N}{A+C}}$</p>
</article></div></main><div class="nav-wrap"><div class="nav"><button class="site-nav"><div class="navicon"></div></button><ul class="nav_items"><li class="nav_item"><a class="nav-page" href="/about/"> About</a></li><li class="nav_item"><a class="nav-page" href="/journal-club/"> Journal Club</a></li><li class="nav_item"><a class="nav-page" href="/publications/"> Publications</a></li><li class="nav_item"><a class="nav-page" href="/project/"> Bachelor Thesis</a></li><li class="nav_item"><a class="nav-page" href="/blogs/"> Blogs</a></li></ul></div><div class="cd-top"><i class="fa fa-arrow-up" aria-hidden="true"></i></div></div><footer id="page_footer"><div class="footer_wrap"><div class="copyright">&copy;2024 - 2025 by Guo Jiayi</div><p style="font-size: 14px">Powered by Academia</p></div></footer><script src="https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/jquery-pjax@latest/jquery.pjax.min.js"></script><script src="/js/main.js"></script></body></html>