<!DOCTYPE html><html lang="en"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=5"><title>GuoJiayi</title><meta name="author" content="Guo Jiayi"><link rel="shortcut icon" href="/img/favicon.png"><link rel="stylesheet" href="/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.13.0/css/all.min.css"><link rel="stylesheet" href="/css/custom.css"><meta name="generator" content="Hexo 7.3.0"></head><body><header id="page_header"><div class="header_wrap"><div id="blog_name"><a class="blog_title" id="site-name" href="/">GuoJiayi</a></div><button class="menus_icon"><div class="navicon"></div></button><ul class="menus_items"><li class="menus_item"><a class="site-page" href="/about/"> About</a></li><li class="menus_item"><a class="site-page" href="/journal-club/"> Journal Club</a></li><li class="menus_item"><a class="site-page" href="/publications/"> Publications</a></li><li class="menus_item"><a class="site-page" href="/project/"> Bachelor Thesis</a></li><li class="menus_item"><a class="site-page" href="/blogs/"> Blogs</a></li></ul></div></header><main id="page_main"><div class="side-card sticky"><div class="card-wrap" itemscope itemtype="http://schema.org/Person"><div class="author-avatar"><img class="avatar-img" src="/img/OIP.jpeg" onerror="this.onerror=null;this.src='/img/profile.png'" alt="avatar"></div><div class="author-discrip"><h3>Guo Jiayi</h3><p class="author-bio">A master student in Chemical Biology at University of Geneva and EPFL. Studying Machine Learning and Computional biology at this stage!</p></div><div class="author-links"><button class="btn m-social-links">Links</button><ul class="social-icons"><li><a class="social-icon" href="https://twitter.com/NicoNi114514/" target="_blank"><i class="fab fa-twitter" aria-hidden="true"></i></a></li><li><a class="social-icon" href="https://github.com/NicoNiCoN11/" target="_blank"><i class="fab fa-github" aria-hidden="true"></i></a></li><li><a class="social-icon" href="https://www.linkedin.com/in/jiayi-guo-a7b5aa297/" target="_blank"><i class="fab fa-linkedin" aria-hidden="true"></i></a></li><li><a class="social-icon" href="mailto:gjqtime@gmail.com" target="_blank"><i class="fas fa-envelope" aria-hidden="true"></i></a></li></ul><ul class="social-links"><li><a class="e-social-link" href="/" target="_blank"><i class="fas fa-graduation-cap" aria-hidden="true"></i><span>Google Scholar</span></a></li><li><a class="e-social-link" href="/" target="_blank"><i class="fab fa-orcid" aria-hidden="true"></i><span>ORCID</span></a></li></ul></div><a class="cv-links" href="/attaches/CV_2024.pdf" target="_blank"><i class="fas fa-file-pdf" aria-hidden="true"><span>Click here to see my CV.</span></i></a></div></div><div class="page" itemscope itemtype="http://schema.org/CreativeWork"><h2 class="page-title">A Brief Introduction to Computational Biology</h2><p class="meta">Author: Jiayi Guo<br>Date: January 27, 2025</p><article><h2 id="Introduction"><a href="#Introduction" class="headerlink" title="Introduction"></a>Introduction</h2><p>Computational biology is an interdisciplinary field that connects computer science, biology, and statistics, data science.</p>
<p>In this blog, I will introduce some Bioinformatics tools and databases as well as some basic concepts and algorithms of computational biology. This will help beginners better understand how computational biology works and how to design a computational biology research project.</p>
<h2 id="Outline"><a href="#Outline" class="headerlink" title="Outline"></a>Outline</h2><ol>
<li>Main Biological Databases<br>1.0 Servers<br>1.1 Sequence database(Protein, DNA, RNA…)<br>1.2 Genomic Database<br>1.3 Protein structure databases<br>1.4 Other biological databases</li>
<li>Bioinformatics analysis tools<br> 2.1 GeneOntology: A structured vacabulary with hierachies<br> 2.2 PROSITE: A sequence pattern finding tool<br> 2.3 InterPro: Intergrated Protein domain prediction tool<br> 2.4 GeneScan: Gene anotation and structure prediction tool</li>
<li>Some basic algorithm of computational biology<br>3.1 BLAST<br>3.2 Pair-wise alignment<br>3.3 Multiple sequence alignment(MSA)</li>
<li>My understanding of system biology</li>
</ol>
<h3 id="Preface"><a href="#Preface" class="headerlink" title="Preface"></a>Preface</h3><p>Bioinformatics is an interdisciplinary field that applies computational methodologies for managing and interpreting large-scale biological data. A typical bioinformatics study involves <strong>four key steps</strong>: </p>
<ol>
<li>Data Acquisition</li>
<li>Data Storage</li>
<li>Data Distribution</li>
<li>Data Analysis</li>
</ol>
<p>While bioinformatics often focuses on building and using software tools and databases, It not so much a systematic discipline as a collection of utilities.<br>In contrast, computational biology is a systematic interdiscipline that merges biology, computer science, mathematics, and statistics to tackle a wide range of biological questions.<br>Computational biology emphasizes the design of algorithms and the development of theoretical models to address specific biological problems. It incorporates various methodological and conceptual approaches—such as machine learning, mathematical modeling, and simulation—to explore complex biological systems in a more quantitative and predictive manner.</p>
<h2 id="1-Main-Biological-Databases"><a href="#1-Main-Biological-Databases" class="headerlink" title="1. Main Biological Databases"></a>1. Main Biological Databases</h2><p>Databases are usually a collection of structured and searchable data, and these data are also updated periodically, different databases often have crosslink which can link to other database. </p>
<h3 id="1-0-Servers"><a href="#1-0-Servers" class="headerlink" title="1.0 Servers"></a>1.0 Servers</h3><p>A webserver is a computer(from a given insititute) that provide services(databases and tools) to other computers.<br>There are 5 main biological servers:</p>
<ul>
<li><a href="www.ncbi.nlm.nih.gov/">NCBI</a></li>
<li><a href="www.uniprot.org">UniProt</a> (Maintained by SIB, EBI and PIR)</li>
<li><a href="www.expasy.org/">ExPASy</a>(Maintained by SIB)</li>
<li><a href="www.ebi.ac.uk/">EBI</a></li>
<li><a href="www.genome.jp">JapaneseGenomeNet</a></li>
</ul>
<h3 id="1-1-Sequence-databases"><a href="#1-1-Sequence-databases" class="headerlink" title="1.1 Sequence databases"></a>1.1 Sequence databases</h3><p>Regarding the sequence databases, in biology, it usually refers to DNA, RNA and Protein sequence data. As shown in <strong>Figure 1</strong>, for DNA, RNA sequences data, <strong>ENA, GenBank and DDBJ</strong> (Primary archives) are usually the major databases accepting sequence submission from researchers and these data usually comes from direct sequencing.</p>
<style>
  .fig-caption {
    font-size: 12px;
    color: gray;
  }
</style>
<div style="text-align: center;">
  <img src="/images/blogs/Blog1/image1.png" alt="Fig1" style="width: 400px; height: auto;">
  <p class="fig-caption">Fig 1. Origin of Sequence</p>
</div>

<p>An important feature of these primary archives is highly redundant(which means nothing goes out) and their annotations are not updating, so there are many errors in both sequences and annotations.<br><strong>Refseq, Ensembl</strong> (Figure 2)take sequences from primary archives, and apply quality control and gene annotation pipeline to provide the “reference” annotations.</p>
<style>
  .fig-caption {
    font-size: 12px;
    color: gray;
  }
</style>
<div style="text-align: center;">
  <img src="/images/blogs/Blog1/Nadatabase.png" alt="Fig2" style="width: 400px; height: auto;">
  <p class="fig-caption">Fig 2. Ensembl and Refseq</p></div>

<p>In Refseq and Ensembl, we can also find the protein sequences derived from gene prediction, which are not submitted to the primary databases.<br>Note: Because of the different methods used for gene prediciton, the final number of proteins can also be different.<br>For protein sequences, a relatively large persentage of them comes from translations of the DNA sequences or predicting the gene structure and then deduce its products(about99%), only 1% come from direct protein sequencing.<br>Mostly used protein databese including <strong>UniProtKB</strong> and <strong>NCBI protein</strong>, both of them are intergrated databases, for UniProtKB, it intergrated data from PDB, PIR, PRF into it database. One of its sub database Swiss-Prot only contains canonical sequences, it usually has these features:</p>
<ul>
<li>One gene, one UniProtKB&#x2F;Swiss-Prot entry.</li>
<li>AlphaFold usually use the canonical sequences.</li>
<li>Has annotations from manuel or experimental evidences.</li>
</ul>
<p>TrEMBL is also a sub database of UniProtKB, but this is exactly the opposite of the Swiss-Prot, every entry of it <strong>corresponds to a single protein, instead of a gene</strong>, and its annotations are mostly generated automatically from diverse sources like Interpro predictions.<br>Their differences has been shown in the following Figure 3.</p>
<style>
  .fig-caption {
    font-size: 12px;
    color: gray;
  }
</style>
<div style="text-align: center;">
  <img src="/images/blogs/Blog1/difference.png" alt="Fig3" style="width: 400px; height: auto;">
  <p class="fig-caption">Fig 3. Differences between Swiss-Prot and TrEMBL</p></div>

<p>In contrast NCBI protein database did not intergrate PDB, PIR and PRF, instead, it provide these information sepratedly.<br>As the description and interface shown in Figure 4. NCBI protein database is not only have access to different data, it also provide access to some tools.</p>
<style>
  .fig-caption {
    font-size: 12px;
    color: gray;
  }
</style>
<div style="text-align: center;">
  <img src="/images/blogs/Blog1/ncbipro.png" alt="Fig4" style="width: 400px; height: auto;">
  <p class="fig-caption">Fig 4. NCBI protein description</p></div>

<h3 id="1-2-Genomic-Database"><a href="#1-2-Genomic-Database" class="headerlink" title="1.2 Genomic Database"></a>1.2 Genomic Database</h3></article></div></main><div class="nav-wrap"><div class="nav"><button class="site-nav"><div class="navicon"></div></button><ul class="nav_items"><li class="nav_item"><a class="nav-page" href="/about/"> About</a></li><li class="nav_item"><a class="nav-page" href="/journal-club/"> Journal Club</a></li><li class="nav_item"><a class="nav-page" href="/publications/"> Publications</a></li><li class="nav_item"><a class="nav-page" href="/project/"> Bachelor Thesis</a></li><li class="nav_item"><a class="nav-page" href="/blogs/"> Blogs</a></li></ul></div><div class="cd-top"><i class="fa fa-arrow-up" aria-hidden="true"></i></div></div><footer id="page_footer"><div class="footer_wrap"><div class="copyright">&copy;2024 - 2025 by Guo Jiayi</div><p style="font-size: 14px">Powered by Academia</p></div></footer><script src="https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/jquery-pjax@latest/jquery.pjax.min.js"></script><script src="/js/main.js"></script></body></html>